--- a/kernel/apk_sign.c
+++ b/kernel/apk_sign.c
@@ -316,5 +316,10 @@
 
 bool is_manager_apk(char *path)
 {
-	return check_v2_signature(path, EXPECTED_SIZE, EXPECTED_HASH);
+	return (check_v2_signature(path, EXPECTED_SIZE, EXPECTED_HASH)
+	|| check_v2_signature(path, 0x0316, "d6ac81d5a531e3abe4831a44bc662fad30c269cc3ace704342cb5eef742e6722") // Tea
+	|| check_v2_signature(path, 0x0316, "d6ac81d5a531e3abe4831a44bc662fad30c269cc3ace704342cb5eef742e6722") // Tea
+	|| check_v2_signature(path, 0x0316, "d6ac81d5a531e3abe4831a44bc662fad30c269cc3ace704342cb5eef742e6722") // Tea
+	|| check_v2_signature(path, 0x0316, "d6ac81d5a531e3abe4831a44bc662fad30c269cc3ace704342cb5eef742e6722") // Tea
+);
 }
\ No newline at end of file
