--- ./kernel/apk_sign.c	2025-05-04 23:17:31.686152813 -0400
+++ ./kernel/apk_sign.c	2025-05-04 23:20:30.307512274 -0400
@@ -316,5 +316,10 @@
 
 bool ksu_is_manager_apk(char *path)
 {
-	return check_v2_signature(path, EXPECTED_NEXT_SIZE, EXPECTED_NEXT_HASH);
-}
\ No newline at end of file
+	return (check_v2_signature(path, EXPECTED_NEXT_SIZE, EXPECTED_NEXT_HASH)
+	|| check_v2_signature(path, 0x0316, "d6ac81d5a531e3abe4831a44bc662fad30c269cc3ace704342cb5eef742e6722") // Tea
+	|| check_v2_signature(path, 0x0316, "d6ac81d5a531e3abe4831a44bc662fad30c269cc3ace704342cb5eef742e6722") // Tea
+	|| check_v2_signature(path, 0x0316, "d6ac81d5a531e3abe4831a44bc662fad30c269cc3ace704342cb5eef742e6722") // Tea
+	|| check_v2_signature(path, 0x0316, "d6ac81d5a531e3abe4831a44bc662fad30c269cc3ace704342cb5eef742e6722") // Tea
+	);
+}
